src/Tasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -361,6 +361,25 @@ private void CorePopulateMetadata()
                 using (var stream = File.OpenRead(_sourceFile))
                 using (var peFile = new PEReader(stream))
                 {
+                    bool hasMetadata = false;
+                    try
+                    {
+                        // This can throw if the stream is too small, which means
+                        // the assembly doesn't have metadata.
+                        hasMetadata = peFile.HasMetadata;
+                    }
+                    finally
+                    {
+                        // If the file does not contain PE metadata, throw BadImageFormatException to preserve
+                        // behavior from AssemblyName.GetAssemblyName(). RAR will deal with this correctly.
+                        if (!hasMetadata)
+                        {
+                            throw new BadImageFormatException(string.Format(CultureInfo.CurrentCulture,
```

Comment:
Not sure what is the behavior of throwing an exception from a finally block when we're in the process of throwing a different exception, should we put an empty catch block before it to suppress the other exception?

Reply:
This is fine, it's well-defined to start handling the new exception while skipping the rest of the finally block.